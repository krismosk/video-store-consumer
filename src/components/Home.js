import React, { Component } from 'react';
import { Route, NavLink, HashRouter} from "react-router-dom";
import Search from "./Search";
import RentalLibrary from "./RentalLibrary";
import CustomerList from "./CustomerList";

class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      movies: [],
      selectedCustomer: "",
    };
  }
  
  // add function
  // fiter function?
  // create rental function

  selectCustomer = (customer) => {
    this.setState({
      selectedCustomer: customer,
    });
  }

  render () {
    return (
      <HashRouter>
        <div className="App">
            <h1>Ada Movie Store</h1>
            <ul className="header">
              <li><NavLink to="/search">Search</NavLink></li>
              <li><NavLink to="/movies">Movies</NavLink></li>
              <li><NavLink to="/customers">Customers</NavLink></li>
              {/* <li><NavLink to="/rentals">Rentals</NavLink></li> */}
            </ul>
            <div className="content">
              <Route path="/search" component={Search}/>
              <Route path="/movies" component={RentalLibrary}/>
              <Route path="/customers" render={() => <CustomerList selectCustomer={this.selectCustomer} />}/>
              {/* <Route path="/rentals" component={Rentals}/> */}
            </div>
          </div>
        </HashRouter>
    );
  }
}

export default Home;