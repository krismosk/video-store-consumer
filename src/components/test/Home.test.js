import React from 'react';
import ReactDOM from 'react-dom';
import Home from '../Home';
import { render, cleanup } from '@testing-library/react';

describe("Home", () => {
  test('it can be rendered and matches the snapshot', () => {
    const { asFragment } = render ( <Home /> );

    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Home />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})