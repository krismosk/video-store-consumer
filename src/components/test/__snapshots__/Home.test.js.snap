// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home it can be rendered and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Ada Movie Store
    </h1>
    <section>
      <div>
        <section>
          <p />
          <p />
          <p />
          <button>
            Checkout Rental
          </button>
        </section>
      </div>
    </section>
    <ul
      class="header"
    >
      <li>
        <a
          href="#/search"
        >
          Search
        </a>
      </li>
      <li>
        <a
          href="#/movies"
        >
          Movies
        </a>
      </li>
      <li>
        <a
          href="#/customers"
        >
          Customers
        </a>
      </li>
    </ul>
    <div
      class="content"
    />
  </div>
</DocumentFragment>
`;
